internal/databases/postgres/backup_push_handler_test.go

```
@@ -0,0 +1,117 @@
+	storeAllCorruptBlocks = storeAllCorruptBlocks || viper.GetBool(internal.StoreAllCorruptBlocksSetting)
+
+	tarBallComposerType := chooseTarBallComposer()
+
+	if deltaFromName == "" {
+		deltaFromName = viper.GetString(internal.DeltaFromNameSetting)
+	}
+	if deltaFromUserData == "" {
+		deltaFromUserData = viper.GetString(internal.DeltaFromUserDataSetting)
+	}
+	if userDataRaw == "" {
+		userDataRaw = viper.GetString(internal.SentinelUserDataSetting)
+	}
+	withoutFilesMetadata = withoutFilesMetadata || viper.GetBool(internal.WithoutFilesMetadataSetting)
+	if withoutFilesMetadata {
+		// files metadata tracking is required for delta backups and copy/rating composers
+		if tarBallComposerType != RegularComposer {
+			tracelog.ErrorLogger.Fatalf(
+				"%s option cannot be used with non-regular tar ball composer",
+				withoutFilesMetadataFlag)
+		}
+		if deltaFromName != "" || deltaFromUserData != "" {
+			tracelog.ErrorLogger.Fatalf(
+				"%s option cannot be used with %s, %s options",
+				withoutFilesMetadataFlag, deltaFromNameFlag, deltaFromUserDataFlag)
+		}
+		tracelog.InfoLogger.Print("Files metadata tracking is disabled")
+		fullBackup = true
+	}
+
+	deltaBaseSelector, err := internal.NewDeltaBaseSelector(
+		deltaFromName, deltaFromUserData, NewGenericMetaFetcher())
+	tracelog.ErrorLogger.FatalOnError(err)
+
+	userData, err := internal.UnmarshalSentinelUserData(userDataRaw)
+	tracelog.ErrorLogger.FatalfOnError("Failed to unmarshal the provided UserData: %s", err)
+
+	folder, err := internal.ConfigureFolder()
+	uploader, err := internal.ConfigureUploaderToFolder(folder)
+
+	arguments := NewBackupArguments(uploader, dataDirectory, utility.BaseBackupPath,
+		permanent, verifyPageChecksums || viper.GetBool(internal.VerifyPageChecksumsSetting),
+		fullBackup, storeAllCorruptBlocks || viper.GetBool(internal.StoreAllCorruptBlocksSetting),
+		tarBallComposerType, NewRegularDeltaBackupConfigurator(deltaBaseSelector),
+		userData, withoutFilesMetadata)
+
```

And here err is not checked too